jablib/src/main/java/org/jabref/logic/util/ExternalLinkCreator.java

```diff
@@ -1,40 +1,152 @@
+
+    private final ImporterPreferences importerPreferences;
+
+    public ExternalLinkCreator(ImporterPreferences importerPreferences) {
+        this.importerPreferences = importerPreferences;
+    }
+
+    /**
+     * Get a URL to the search results of Google Scholar for the BibEntry's title
+     *
+     * @param entry The entry to search for. Expects the BibEntry's title to be set for successful return.
+     * @return The URL if it was successfully created
+     */
+    public Optional<String> getGoogleScholarSearchURL(BibEntry entry) {
+        return entry.getField(StandardField.TITLE).flatMap(title -> {
+            String baseUrl = importerPreferences.getSearchEngineUrlTemplates()
+                                                .getOrDefault("Google Scholar", DEFAULT_GOOGLE_SCHOLAR_SEARCH_URL);
+            Optional<String> author = entry.getField(StandardField.AUTHOR);
+            return buildSearchUrl(baseUrl, DEFAULT_GOOGLE_SCHOLAR_SEARCH_URL, title, author, "Google Scholar");
+        });
+    }
 
     /**
      * Get a URL to the search results of ShortScience for the BibEntry's title
      *
      * @param entry The entry to search for. Expects the BibEntry's title to be set for successful return.
      * @return The URL if it was successfully created
      */
-    public static Optional<String> getShortScienceSearchURL(BibEntry entry) {
-        return entry.getField(StandardField.TITLE).map(title -> {
-            URIBuilder uriBuilder;
-            try {
-                uriBuilder = new URIBuilder(SHORTSCIENCE_SEARCH_URL);
-            } catch (URISyntaxException e) {
-                // This should never be able to happen as it would require the field to be misconfigured.
-                throw new AssertionError("ShortScience URL is invalid.", e);
+    public Optional<String> getShortScienceSearchURL(BibEntry entry) {
+        return entry.getField(StandardField.TITLE).flatMap(title -> {
+            String baseUrl = importerPreferences.getSearchEngineUrlTemplates()
+                                                .getOrDefault("Short Science", DEFAULT_SHORTSCIENCE_SEARCH_URL);
+            Optional<String> author = entry.getField(StandardField.AUTHOR);
+            return buildSearchUrl(baseUrl, DEFAULT_SHORTSCIENCE_SEARCH_URL, title, author, "ShortScience");
+        });
+    }
+
+    /**
+     * Builds a search URL using either template replacement or query parameters
+     *
+     * @param baseUrl The custom or default base URL
+     * @param defaultUrl The fallback default URL
+     * @param title The title to search for
+     * @param author Optional author to include in search
+     * @param serviceName Name of the service for logging
+     * @return Optional containing the constructed URL, or empty if construction failed
+     */
+    private Optional<String> buildSearchUrl(String baseUrl, String defaultUrl, String title, Optional<String> author, String serviceName) {
```

Comment:
I don't like `Optional` as parameter, better `@Nullable` author.

But thats a general decision of our code...  Yesterday, I coded with `null`.